### ISSUE TITLE
Key verification events can be in the timeline as well.

### ISSUE THREAD

**Issue by poljar** [ruma/ruma-events#33]
Key verification events as per https://github.com/matrix-org/matrix-doc/pull/2241 will be sent out as normal room messages. This means they will contain a `sender`, `event_id`, `origin_server_ts` etc.

Key verification events need to be updated to include those attributes.

The affected event types are:
- m.key.verification.accept
- m.key.verification.cancel
- m.key.verification.key
- m.key.verification.mac
- m.key.verification.request
- m.key.verification.start

**Comment by jplatte** [comment]
@poljar Do you want to take a stab at implementing this? It should be a straight-forward change (move the modules to `room/`, declare them as room events like the other ones, no changes to the declared fields necessary) if I properly understand this issue.

**Comment by poljar** [comment]
I'm not sure that just moving them to room events is enough. They can still be a to-device event. But sure, I'll check it out.

**Comment by poljar** [comment]
So checked it out. The verification events are implemented using the `ruma_event!` macro. The macro takes an argument for the kind of the event. 

Changing this from `Event` to `RoomEvent` will make the `event_id`, `sender` and the rest appear, but the event will probably fail to be de-serialized if it's send as a to-device event.

So we could:
1. Create a new kind in the macro
2. Define the struct fields manually
3. Create two key verification event implementations one for each event kind

**Comment by jplatte** [comment]
I don't understand. Are these events sometimes room events and sometimes not?

**Comment by poljar** [comment]
Yeah, when they were first implemented they were sent using the [sendToDevice](https://matrix.org/docs/spec/client_server/latest#put-matrix-client-r0-sendtodevice-eventtype-txnid) endpoint, those events aren't part of the DAG nor a room, and aren't kept around in the history.

The linked spec proposal adds the ability to send those events using [rooms/{room_id}/send](https://matrix.org/docs/spec/client_server/latest#put-matrix-client-r0-rooms-roomid-send-eventtype-txnid) API endpoint, making them normal room events.

For backwards compatibility the first method of sending those events won't be dropped.

**Comment by jplatte** [comment]
Okay so I understand this is a proposal, nothing final? In that case, we probably shouldn't do anything yet, but it's good to keep this on our radar nonetheless. I think having separate event types for the room events and send-to-device events would be the way to go once this lands in the next version of the spec.

**Comment by poljar** [comment]
While it hasn't been merged yet, it's very likely to be final. It has been implement and merged in the [js-skd](https://github.com.cnpmjs.org/matrix-org/matrix-js-sdk/pull/1050).

I agree that we don't need to rush this out, I will probably need this in a couple of months.
